ios/PlayerModule.swift

```diff
@@ -45,7 +45,18 @@ class PlayerModule: NSObject, RCTBridgeModule {
             else {
                 return
             }
-            self?.players[nativeId] = PlayerFactory.create(playerConfig: playerConfig)
+            let player = PlayerFactory.create(playerConfig: playerConfig)
+            self?.players[nativeId] = player
+
+            playerConfig.remoteControlConfig.prepareSource = { [weak self] type, sourceConfig in
+                guard let sourceModule = self?.bridge[SourceModule.self],
+                      let sourceNativeId = sourceModule.nativeId(where: { $0.sourceConfig === sourceConfig }),
+                      let castSourceConfig = sourceModule.retrieveCastSourceConfig(sourceNativeId) else {
+                    return nil
+                }
+
+                return castSourceConfig
+            }
```

Comment:
We have to look up the cast-specific source config on the fly and we have to use `sourceConfig` to do that, as `player.source` returns `nil` when this is called.